services/ai_client.py

```
@@ -0,0 +1,105 @@
+            if response.status_code != 200:
+                raise AIClientError(f"API request failed with status {response.status_code}: {response.text}")
+            
+            data = response.json()
+            
+            if "result" not in data or "response" not in data["result"]:
+                raise AIClientError(f"Unexpected API response format: {data}")
+            
+            return data["result"]["response"]
+            
+        except requests.exceptions.Timeout:
+            raise AIClientError(f"Request timed out after {self.timeout_ms}ms")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)